Continuing training at iteration -1
Top contributors:
Collecting rollouts (-1)...
Training network...
wandb: WARNING Step cannot be set when using syncing with tensorboard. Please log your step values as a metric such as 'global_step'
Top contributors:
	ML-bot: 1000324
Collecting rollouts (0)...
Training network...
Top contributors:
	ML-bot: 1004229
Collecting rollouts (1)...
Training network...
Top contributors:
	ML-bot: 1005534
Collecting rollouts (2)...
Training network...
Top contributors:
	ML-bot: 996560
Collecting rollouts (3)...
Training network...
Top contributors:
	ML-bot: 1007373
Collecting rollouts (4)...
Training network...
Top contributors:
	ML-bot: 995331
Collecting rollouts (5)...
Training network...
Top contributors:
	ML-bot: 1003502
Collecting rollouts (6)...
Training network...
Top contributors:
	ML-bot: 997972
Collecting rollouts (7)...
Training network...
Top contributors:
	ML-bot: 1003982
Collecting rollouts (8)...
Training network...
Top contributors:
	ML-bot: 996954
Collecting rollouts (9)...
Training network...
Top contributors:
	ML-bot: 1000326
Collecting rollouts (10)...
Training network...
Top contributors:
	ML-bot: 1001280
Collecting rollouts (11)...
Training network...
Top contributors:
	ML-bot: 1003972
Collecting rollouts (12)...
Training network...
Top contributors:
	ML-bot: 999106
Collecting rollouts (13)...
Training network...
Top contributors:
	ML-bot: 1004706
Collecting rollouts (14)...
Training network...
Top contributors:
	ML-bot: 1003121
Collecting rollouts (15)...
Training network...
Top contributors:
	ML-bot: 998042
Adding model to pool...
Saving model...
Collecting rollouts (16)...
Training network...
Top contributors:
	ML-bot: 1006517
Collecting rollouts (17)...
Training network...
Top contributors:
	ML-bot: 998749
Collecting rollouts (18)...
Training network...
Top contributors:
	ML-bot: 1001127
Collecting rollouts (19)...
Training network...
Top contributors:
	ML-bot: 1006534
Collecting rollouts (20)...
Training network...
Top contributors:
	ML-bot: 996971
Collecting rollouts (21)...
Training network...
Top contributors:
	ML-bot: 1000846
Collecting rollouts (22)...
Training network...
Top contributors:
	ML-bot: 1002122
Collecting rollouts (23)...
Training network...
Top contributors:
	ML-bot: 1006197
Collecting rollouts (24)...
Training network...
Top contributors:
	ML-bot: 1000560
Collecting rollouts (25)...
Training network...
Top contributors:
	ML-bot: 998115
Collecting rollouts (26)...
Training network...
Top contributors:
	ML-bot: 1002299
Collecting rollouts (27)...
Training network...
Top contributors:
	ML-bot: 1004038
Collecting rollouts (28)...
Training network...
Top contributors:
	ML-bot: 998000
Collecting rollouts (29)...
Training network...
Top contributors:
	ML-bot: 1001474
Collecting rollouts (30)...
Training network...
Top contributors:
	ML-bot: 1000444
Collecting rollouts (31)...
Traceback (most recent call last):
  File "c:/Users/manue/Downloads/Advanced ML Rocket League-bot project/learner.py", line 186, in <module>
    alg.run(iterations_per_save=10, save_dir="save")
  File "c:\Users\manue\Downloads\Advanced ML Rocket League-bot project\rocket_learn\ppo.py", line 165, in run
    self.calculate(_iter(), iteration)
  File "c:\Users\manue\Downloads\Advanced ML Rocket League-bot project\rocket_learn\ppo.py", line 254, in calculate
    for buffer in buffers:  # Do discounts for each ExperienceBuffer individually
  File "c:\Users\manue\Downloads\Advanced ML Rocket League-bot project\rocket_learn\ppo.py", line 157, in _iter
    rollout = next(rollout_gen)
  File "c:\Users\manue\Downloads\Advanced ML Rocket League-bot project\rocket_learn\rollout_generator\redis\redis_rollout_generator.py", line 161, in generate_rollouts
    data = self.redis.blpop(ROLLOUTS)[1]
  File "C:\Users\manue\AppData\Local\Programs\Python\Python37\lib\site-packages\redis\commands\core.py", line 2473, in blpop
    return self.execute_command("BLPOP", *keys)
  File "C:\Users\manue\AppData\Local\Programs\Python\Python37\lib\site-packages\redis\client.py", line 1242, in execute_command
    lambda error: self._disconnect_raise(conn, error),
  File "C:\Users\manue\AppData\Local\Programs\Python\Python37\lib\site-packages\redis\retry.py", line 46, in call_with_retry
    return do()
  File "C:\Users\manue\AppData\Local\Programs\Python\Python37\lib\site-packages\redis\client.py", line 1240, in <lambda>
    conn, command_name, *args, **options
  File "C:\Users\manue\AppData\Local\Programs\Python\Python37\lib\site-packages\redis\client.py", line 1215, in _send_command_parse_response
    return self.parse_response(conn, command_name, **options)
  File "C:\Users\manue\AppData\Local\Programs\Python\Python37\lib\site-packages\redis\client.py", line 1254, in parse_response
    response = connection.read_response()
  File "C:\Users\manue\AppData\Local\Programs\Python\Python37\lib\site-packages\redis\connection.py", line 824, in read_response
    response = self._parser.read_response(disable_decoding=disable_decoding)
  File "C:\Users\manue\AppData\Local\Programs\Python\Python37\lib\site-packages\redis\connection.py", line 318, in read_response
    raw = self._buffer.readline()
  File "C:\Users\manue\AppData\Local\Programs\Python\Python37\lib\site-packages\redis\connection.py", line 249, in readline
    self._read_from_socket()
  File "C:\Users\manue\AppData\Local\Programs\Python\Python37\lib\site-packages\redis\connection.py", line 192, in _read_from_socket
    data = self._sock.recv(socket_read_size)
KeyboardInterrupt